src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -581,7 +581,14 @@ export class NotebookModel extends Disposable implements INotebookModel {
 		if (cell) {
 			let index = this.findCellIndex(cell);
 			if (index > -1) {
-				cell.cellType = cell.cellType === CellTypes.Markdown ? CellTypes.Code : CellTypes.Markdown;
+				// If cell is a markdown cell, transform into code cell (and vice versa)
+				if (cell.cellType === CellTypes.Markdown) {
+					cell.cellType = CellTypes.Code;
+					// Ensure override language is reset
+					cell.setOverrideLanguage('');
```

Comment:
Do you not want to reset the override language no matter what? I would assume it wouldn't be applicable for the new cell type regardless

Reply:
good catch. makes sense